src/main/core/manager.rs

```diff
@@ -506,6 +512,8 @@ impl<'a> Manager<'a> {
             worker::with_global_object_counters(|alloc_counter, dealloc_counter| {
                 log::info!("Global allocated object counts: {}", alloc_counter);
                 log::info!("Global deallocated object counts: {}", dealloc_counter);
+                sim_stats.objects.alloc_counts = alloc_counter.clone();
+                sim_stats.objects.dealloc_counts = dealloc_counter.clone();
```

Comment:
Super super optional nit: I personally prefer to keep internal members private and use `set()` methods, but I think that's just a style thing. It does require more boilerplate for the set methods.

Reply:
I left this as public struct fields for now, which I think help keeps the boilerplate code down for these "collection" objects, but we can always change it later.